gozo-stores/src/pages/ProductListing.jsx

```diff
@@ -18,14 +20,14 @@ export default function ProductListing() {
                 <div class="product-listing-area">
 
                     <div class="heading">
-                        <h4>Showing all products</h4><p>(Showing  products)</p>
+                        <h4>Showing all products</h4><p>(Showing {products.length}  products)</p>
                     </div>
 
                     <div class="listed-products"> 
 
                         {loading? <p> Loading Products </p> : 
                             <div class="normal-products">
-                                
+                                {products.map(({name, src, finalPrice, rating}) => <Card name={name} src={src} price={finalPrice} rating={rating} />)}
```

Comment:
Just a suggestion, Instead of destructuring it over here, you can pass the product and then destructure it in the Card Component.

Reply:
done